stock_quant_manual_assign/wizard/assign_manual_quants.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2015 Mikel Arregi - AvanzOSC
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# Copyright 2018 Fanha Giang
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare, float_is_zero
+
+
+class AssignManualQuants(models.TransientModel):
+    _name = "assign.manual.quants"
+    _description = "Assign Manual Quants"
+
+    @api.constrains("quants_lines")
+    def _check_qty(self):
+        precision_digits = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        for record in self.filtered("quants_lines"):
+            if (
+                float_compare(
+                    record.lines_qty,
+                    record.move_id.product_qty,
+                    precision_digits=precision_digits,
+                )
+                > 0
+            ):
+                raise ValidationError(_("Quantity is higher than the needed one"))
+
+    @api.depends("move_id", "quants_lines", "quants_lines.qty")
+    def _compute_qties(self):
+        for record in self:
+            record.lines_qty = sum(
+                record.quants_lines.filtered("selected").mapped("qty")
+            )
+            record.move_qty = record.move_id.product_qty - record.lines_qty
+
+    lines_qty = fields.Float(
+        string="Reserved qty",
+        compute="_compute_qties",
+        digits="Product Unit of Measure",
+    )
+    move_qty = fields.Float(
+        string="Remaining qty",
+        compute="_compute_qties",
+        digits="Product Unit of Measure",
+    )
+    quants_lines = fields.One2many(
+        comodel_name="assign.manual.quants.lines",
+        inverse_name="assign_wizard",
+        string="Quants",
+    )
+    move_id = fields.Many2one(
+        comodel_name="stock.move",
+        string="Move",
+    )
+
+    def assign_quants(self):
+        move = self.move_id
+        self.move_id.move_line_ids.unlink()
+        for line in self.quants_lines:
+            line._assign_quant_line()
+        if move.picking_type_id.auto_fill_qty_done:
+            if move.from_immediate_transfer:
+                move.quantity_done = self.lines_qty
+            else:
+                # Auto-fill all lines as done
+                for ml in move.move_line_ids:
+                    ml.qty_done = ml.reserved_uom_qty
```

Comment:
There is no `auto_fill_qty_done` anymore and the distinction between immediate and planned transfers is eliminated from odoo17. Then, qty done and reserved qty are merged into one column as `quantity` field.
So, I think we can remove this part.